src/validation.h

```diff
@@ -495,6 +495,7 @@ class CChainState
     node::BlockManager& m_blockman;
 
     /** Chain parameters for this chainstate */
+    /* TODO: replace with m_chainman.GetParams() */
```

Comment:
By this, do you mean all references to `(CChainState::)m_params` by `CChainState` member functions should instead be references to `m_chainman.GetParams()`?

Reply:
Yeah. Or perhaps add a `GetParams()` and `GetConsensus()` to `CChainState` that just call `m_chainman.GetParams()` or `m_chainman.GetConsensus()`.

Reply:
(I didn't start on this here since I imagine it would have a bunch of conflicts with other PRs)

Reply:
Sounds good!